src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/StopAzureVMCommand.cs

```diff
@@ -31,6 +32,7 @@ public class StopAzureVMCommand : VirtualMachineActionBaseCmdlet
            Position = 1,
            ValueFromPipelineByPropertyName = true,
            HelpMessage = "The virtual machine name.")]
+        [ResourceNameCompleter("Microsoft.Compute/virtualMachines", "ResourceGroupName")]
```

Comment:
CAn you also change [this base class](https://github.com/Azure/azure-powershell/blob/b2c9eead74d69b12aae236ca9102a36697928a0d/src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs#L43) to pull out the name from thre resourceId as well... most of the child classes of this class have the same issue.. they require a name param even when a resourceId is specified.

Reply:
Resolved offline, will be addressed in a different PR.